src/Tools/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/SolutionState.cs

```diff
@@ -0,0 +1,259 @@
+        /// </summary>
+        public List<DiagnosticResult> ExpectedDiagnostics { get; } = new List<DiagnosticResult>();
+
+        /// <summary>
+        /// Gets or sets a value indicating whether markup can be used to identify diagnostics within expected inputs
+        /// and outputs. The default value is <see cref="MarkupMode.Allow"/> for original sources or
+        /// <see cref="MarkupMode.IgnoreFixable"/> for fixed sources.
+        /// </summary>
+        /// <remarks>
+        /// <para>Diagnostics expressed using markup are combined with explicitly-specified expected diagnostics.</para>
+        ///
+        /// <para>Supported markup syntax includes the following:</para>
+        ///
+        /// <list type="bullet">
+        /// <item><description><c>[|text|]</c>: indicates that a diagnostic is reported for <c>text</c>. The diagnostic
+        /// descriptor is located via <see cref="AnalyzerTest{TVerifier}.GetDiagnosticAnalyzers"/>. This syntax may only
+        /// be used when the first analyzer provided by <see cref="AnalyzerTest{TVerifier}.GetDiagnosticAnalyzers"/>
+        /// supports a single diagnostic.</description></item>
+        /// <item><description><c>{|ID1:text|}</c>: indicates that a diagnostic with ID <c>ID1</c> is reported for
+        /// <c>text</c>. The diagnostic descriptor for <c>ID1</c> is located via
+        /// <see cref="AnalyzerTest{TVerifier}.GetDiagnosticAnalyzers"/>. If no matching descriptor is found, the
+        /// diagnostic is assumed to be a compiler-reported diagnostic with the specified ID and severity
+        /// <see cref="DiagnosticSeverity.Error"/>.</description></item>
+        /// </list>
+        /// </remarks>
+        public MarkupMode AllowMarkup { get; set; }
+
+        /// <summary>
+        /// Applies the <see cref="InheritanceMode"/> using a specified base state.
+        /// </summary>
+        /// <remarks>
+        /// <para>This method evaluates <see cref="AdditionalFilesFactories"/>, and places the resulting additional
+        /// files in the <see cref="AdditionalFiles"/> collection of the result before returning.</para>
+        /// </remarks>
+        /// <param name="baseState">The base state to inherit from, or <see langword="null"/> if the current state is
+        /// the root state.</param>
+        /// <param name="fixableDiagnostics">The set of diagnostic IDs to treat as fixable. Fixable diagnostics present
+        /// in the <see cref="ExpectedDiagnostics"/> collection of the base state are never inherited.</param>
+        /// <returns>A new <see cref="SolutionState"/> representing the current state with inherited values applied
+        /// where appropriate. The <see cref="InheritanceMode"/> of the result is
+        /// <see cref="StateInheritanceMode.Explicit"/>.</returns>
+        public SolutionState ApplyInheritedState(SolutionState baseState, ImmutableArray<string> fixableDiagnostics)
```

Comment:
💭 Perhaps `WithInheritedValuesApplied` to match the naming of `WithProcessedMarkup`?

Reply:
➡️ Renamed to `WithInheritedValuesApplied`